### PR TITLE
DExx work

### PR DESCRIPTION
**Pull request by the-snowwhite**
Add sd-image getting started guide for the DE0_Nano_SoC  and DE10_Nano

### CODE REVIEW

**Comment by cerna** [comment]
Nice, finally some documentation for this.

Just - not trying to be nitpick - but how come the BeagleBone images keep the `/machinekit-images` URL? And not `/machinekit-images-for-bbb`.

Also, for SSJ (Simple Stupid Joe) who will look at premade images and will see mksocfpga, there is nothing general about what it even is in the Machinekit documentation. This is made for someone with a programming background. (Just saying.)

**Comment by the-snowwhite** [comment]
@cerna Thanks I added a commit for the url mishap.
About the mksocfpga:
I tried my best to avoid using this term as the sd-images only cover the Altera/Intel part and not Xilinx (zynq)

A further complication is that the release is a generic release for a wide range of soc boards, only customization being the default user name is  machinekit:

[DEx(x) Mksocfpga Debian Stretch may 2019](https://github.com/the-snowwhite/soc-image-buildscripts/releases/tag/2.1)

Too much info ?

 some of this can be remedied by placing a google search link like this:

    link:https://encrypted.google.com/search?q=Mksocfpga [What is Mksocfpga ]
[What is Mksocfpga](https://encrypted.google.com/search?q=Mksocfpga)
or

    link:https://github.com/machinekit/mksocfpga/blob/master/README.MD [What is Mksocfpga ]
[What is Mksocfpga](https://github.com/machinekit/mksocfpga/blob/master/README.MD)
or

    link:https://github.com/machinekit/mksocfpga [What is Mksocfpga ]
[What is Mksocfpga](https://github.com/machinekit/mksocfpga)

somewhere ?

File: docs/getting-started/machinekit-de10-images.asciidoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+:skip-front-matter:
+
+= DEx(x) Debian link:https://github.com/machinekit/mksocfpga[ Mksocfpga ] Images
+
+link:https://encrypted.google.com/search?q=Mksocfpga[What is Mksocfpga ]
+
+There are pre-built (debian Stretch) sd-card images to get started with Machinekit on the
+SocFPGA based DE0_Nano_SoC, DE10_Nano and the DE1_SoC boards from Terasic
+
+- link:http://deb.machinekit.io/uploads/de10-nano/[Sd-card inages are here]
```
Comment (ArcEye) [inline docs/getting-started/machinekit-de10-images.asciidoc]: Typo - 'inages' for 'images'
Otherwise all seems good.

Comment (the-snowwhite) [reply docs/getting-started/machinekit-de10-images.asciidoc]: OK

**Comment by cerna** [comment]
@the-snowwhite 
It's just, I have been frequenting the LinuxCNC forum and the majority of discussion is of the "Can I use the 7I92 with 7I77", "When will the 7I97 be available" or "What PC to pair with 7I92" variety. This is a very nice project which paired with some simple addon card with couple cheap H-bridges for 0-10V analog signal and input protection/opamp differential comparators for could be a valid alternative for low-cost projects. Problem is, not many users are willing to invest the energy and time to truly deeply understand it.

It needs a simple "Idiot's guide" with reference project which everybody can follow. It would be such a waste if only "[hipsters](https://forum.linuxcnc.org/24-hal-components/36131-run-back-end-and-front-end-on-different-machines?start=10#131336)" and one company (I know of) would use this.

**Comment by luminize** [comment]
> On 14 May 2019, at 07:57, cerna <notifications@github.com> wrote:
> 
> @the-snowwhite
> It's just, I have been frequenting the LinuxCNC forum and the majority of discussion is of the "Can I use the 7I92 with 7I77", "When will the 7I97 be available" or "What PC to pair with 7I92" variety. This is a very nice project which paired with some simple addon card with couple cheap H-bridges for 0-10V analog signal and input protection/opamp differential comparators for could be a valid alternative for low-cost projects. Problem is, not many users are willing to invest the energy and time to truly deeply understand it.
> 
> It needs a simple "Idiot's guide" with reference project which everybody can follow. It would be such a waste if only "hipsters" and one company (I know of) would use this.
> 

Hi @cerna
The hard part is that for an idiots guide to work, you need idiot hardware and an idiot application, and idiots who are willing to spend a minimum amount of effort and money... (both not much, but they are required)

I would be willing to spend money on a fully assembled de0 breakout board that Charles’ designed some years ago. I soldered one by hand and although that was not technically hard, it was time consuming.

I guess I’m trying to say that you’re right, and that for something to get traction, all the bits need to be in place. Like the success of the beaglebone images with cramps/bebopr board a few years ago.

I’d love some basic reference thru hole (solderable) electrical designs which can be mixed/matched, put in a kicad design, order PCB and parts, and solder.

Is this something worthwhile discussing on the google group forum you think?

**Comment by the-snowwhite** [comment]
@cerna 
@luminize 
I find it importaint to split this discussion between:
Users and developers as the "getting started for dummies guide" for both of these groups are
different.

> It needs a simple "Idiot's guide" with reference project which everybody can follow

I'ts a bit like the hen and egg parable:
The "idots guide" named here would be for a Machinekit Mksocfpga "user", made for somthing a Machinekit Mksocfpga "Developer" has created, not many developers have been able to "step up to the plate", and this is gratly due to missing docs.

A potential Mksocfpga developer would be asking quistions like in this forum thread:
[DE10 Nano suggested development environment](https://groups.google.com/forum/#!topic/machinekit/eVhvTnuhblE)

And I'm trying to make an effort to have somthing to link to in the Mk Docs that can pull designers of new stuff in, provide basic how to get started/how things work (for Developers) and facilitate getting them to commit their "new additions" back to Machinekit after sucessfull creation.
I have a loose sketch here:
[Mksocfpga Developer:](https://gist.github.com/the-snowwhite/65ede1ec7a4530035164efcaa27df3a2)

**Pull request merged**